docs/DeveloperGuide.md

```diff
@@ -1,18 +1,616 @@
+
+### Features
+#### Main Menu
+- Loading up user data
+- Access to various features
+- Saving outstanding user data to respective save files
+
+#### Feature 1: Manual Tracker & Entry Tracker
+##### Overview
+__Ledgers and Entries__
+
+In this feature, we represent the transactions incurred by the users as ```Entry``` instances.
+Instances of ```Entry``` class are categorised by the date of origin, which is represented by
+```Ledger``` instances.
+
+```Entry``` instances are characterized by the following: 
+* Time of transaction
+* Type of transaction: Income/ Expense 
+* Amount in transaction
+* Category of spending/ expenditure
+* Description
+
+```Ledger``` instances are characterized by the following: 
+* Time of transaction
+* Collection of ```Entry```instances
+
+##### Manual Tracker
+
+The Manual Tracker is a feature that allows users to manage Ledgers with create, delete
+and open operations. Ledgers is a class that maintains a list of transactions that are 
+recorded for a given date. 
+
+The Entry Tracker is fundamentally similar to the Manual Tracker, except it manages ```Entry``` instances
+instead of ```Ledger```. Entry Tracker is initialized when a ```Ledger``` instance is "opened", whereby 
+the Entry Tracker facilitate the manipulation of the collection of ```Entry``` instances that are associated with
+that particular ```Ledger``` instance.
+
+For the sake of brevity, this section will focus on the discussion of the Manual Tracker.
+
+The Manual Tracker is capable of executing the following states of operation:
+
+|States| Operations | 
+|--------|----------|
+|```MAIN_MENU```|Go to main menu for users to choose the available operations
+|```CREATE_LEDGER```|Create a ledger specified by date, and append it to ```ledgerList```.
+|```DELETE_LEDGER```|Delete an existing ledger, referenced by date or index.
+|```OPEN_LEDGER```|Go to subroutine "Entry Tracker" for the entries recorded  under the specified ledger.
+
+#### Architecture
+
+##### Architecture Overview
+
+![](uml_images/manualTracker/images/Architecture_ManualTracker.png)
+
+
+|Module| Function | 
+|--------|----------|
+| ```Parser```|Parse inputs from user and return ```CommandPacket``` instance with organised ```commandString``` and ```paramMap``` 
+| ```Tracker/ Handler```|Manages the overall workflow of the Manual Tracker; identifies operation required from input and executes the corresponding ```command```.
+| ```Data``` |Refers to ```Ledger``` instances, stores relevant data of the day's transactions.
+| ```Data List``` |Refers to ```LedgerList``` instances, maintains Ledger instances within the program. 
+| ```Commands``` |Processes information from ```CommandPacket``` and executes the appropriate process from recognised params.
+| ```Logic``` |Outlines the abstract behavior of commands, as well as handle verification of params with appropriate error handling.
+
+
+
+##### Command and Logic
+
+![](uml_images/manualTracker/images/Commands_Logic_edited.png)
```

Comment:
Regarding this class diagram,  based on the dashed line from the various commands to the ParamChecker, I am assuming that the command classes have dependencies with the ParamChecker. In this case, is showing multiplicities for dependencies correct? Perhaps you might want to consider removing it.  You can also consider this change for the class diagram under the handler and parser section. Perhaps you may also want to consider capitalising the first letter of the command classes. Also, I am not too sure if the current notation of singleton is correct or not.

![image](https://user-images.githubusercontent.com/60005925/97535953-63634a00-19f7-11eb-9cf6-c01d556ec5bf.png)